value/src/main/java/com/google/auto/value/processor/AutoValueIdsProcessor.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright (C) 2014 Google, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.auto.value.processor;
+
+import com.google.auto.common.MoreElements;
+import com.google.auto.common.SuperficialValidation;
+import com.google.auto.service.AutoService;
+import com.google.auto.value.AutoValue;
+import com.google.common.collect.ImmutableSet;
+
+import javax.annotation.processing.AbstractProcessor;
+import javax.annotation.processing.Processor;
+import javax.annotation.processing.RoundEnvironment;
+import javax.lang.model.SourceVersion;
+import javax.lang.model.element.Element;
+import javax.lang.model.element.TypeElement;
+import javax.tools.Diagnostic;
+import java.util.Set;
+
+import static com.google.auto.common.MoreElements.isAnnotationPresent;
+
+/**
+ * Annotation processor that checks that the type that {@link com.google.auto.value.AutoValue.Id}
+ * is applied to is nested inside an {@code @AutoValue} class. The actual code generation for ids
+ * is done in {@link AutoValueProcessor}.
+ *
+ * @author Rafael Torres
+ */
+@AutoService(Processor.class)
+public class AutoValueIdsProcessor extends AbstractProcessor {
```

Comment:
I was not sure if I should create another processor to validate or if I should use AutoValueBuilderProcessor. My suggestion is to rename AutoValueBuilderProcessor to some generic name and use it to validate Builder, Validate and Id.